iscsi/iscsi.go

```diff
@@ -119,8 +111,8 @@ func parseSessions(lines string) []iscsiSession {
 }
 
 // sessionExists checks if an iSCSI session exists
-func sessionExists(tgtPortal, tgtIQN string) (bool, error) {
-	sessions, err := getCurrentSessions()
+func sessionExists(logger *logr.Logger, tgtPortal, tgtIQN string) (bool, error) {
```

Comment:
You can probably remove the logr import by using klog.Logger instead of logr.Logger, it's a type alias.

Note that logr.Logger is meant to be passed by value. Copying it is cheap. A pointer is used when passing a logger is optional, which is not the case here (no nil check before using it).